codewiki/src/be/dependency_analyzer/analysis/repo_analyzer.py

```
@@ -40,7 +66,46 @@ def analyze_repository_structure(self, repo_dir: str) -> Dict:
+
+        gitignore_file = os.path.join(self.repo_path, ".gitignore")
+        if os.path.exists(gitignore_file):
+            with open(gitignore_file, "r", encoding="utf-8") as f:
+                lines = f.readlines()
+            self.gitignore_spec = PathSpec.from_lines(GitWildMatchPattern, lines)
+        else:
+            self.gitignore_spec = None
+
+    def _check_git_availability(self) -> bool:
+        try:
+            result = subprocess.run(["git", "--version"], capture_output=True, timeout=5)
```

Ruff (0.14.13)

Starting a process with a partial executable path

(S607)